extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -229,6 +369,70 @@ class MultipleBackendTask extends ReplicateObject {
         });
     }
 
+    /**
+     * Get byte ranges for an object of the given content length, such that the
+     * range count does not exceed 1024 elements.
+     * @param {Number} contentLength - The content length of the whole object
+     * @return {Array} The array of byte ranges.
+     */
+    _getRanges(contentLength) {
+        const ranges = [];
+        let start = 0;
+        let end = 0;
+        for (let i = 0; i < MPU_GCP_MAX_PARTS; i++) {
```

Comment:
Since you rely on the condition `end >= contentLength - 1` in the loop to break from it (which is fine for me), you may just do an infinite loop `while` condition instead, simplifying the reasoning.

Then with the same logic, for code clarity you could also rewrite the loop as a `while (end < contentLength - 1)` and adding the last part if not empty after the loop.